temporalio/worker/_workflow.py

```diff
@@ -193,6 +206,47 @@ async def _handle_activation(
     ) -> None:
         global LOG_PROTOS
 
+        def get_traceback_for_thread(
+            thread_id: int,
+        ) -> Tuple[Optional[TracebackType], bool]:
+            """Take a thread id and attempt to construct a traceback from its frames.
+
+            Args:
+                thread_id: Thread ID to be checked.
+
+            Returns:
+                A tuple containing:
+                    0: A TracebackType if the supplied thread id is valid, None otherwise.
+                    1: A bool that determines whether the error message should warn about a possibly incomplete traceback.
+            """
+            # retrieve all frames
+            frame = sys._current_frames().get(thread_id)
+
+            # if the id is not present, the thread must have terminated or there is another error
+            if not frame:
+                return (None, False)
+
+            top_tb = TracebackType(
+                None, tb_frame=frame, tb_lasti=frame.f_lasti, tb_lineno=frame.f_lineno
+            )
+
+            # Construct tb stack, might be a better way/builtin to do this
+            tb = top_tb
+            try:
+                while frame.f_back:
+                    frame = frame.f_back
+                    tb.tb_next = TracebackType(
+                        None,
+                        tb_frame=frame,
+                        tb_lasti=frame.f_lasti,
+                        tb_lineno=frame.f_lineno,
+                    )
+                    tb = tb.tb_next
```

Comment:
@twin-drill and I discussed this in another channel -- it's returning the reverse of the standard Python order currently. And also a frame for `_handle_activation` is appearing at the start of the traceback, but that's the frame that spawns the thread.